Experimental/MultiplayerUseCases/Assets/Common/Scripts/NetworkManagerUI.cs

```diff
@@ -1,76 +1,93 @@
-using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Unity.Netcode.Transports.UTP;
 using UnityEngine;
+using UnityEngine.EventSystems;
 using UnityEngine.SceneManagement;
-using UnityEngine.UI;
+using UnityEngine.UIElements;
 
 namespace Unity.Netcode.Samples.MultiplayerUseCases.Common
 {
-
+    [RequireComponent(typeof(UIDocument))]
     public class NetworkManagerUI : MonoBehaviour
     {
-        [SerializeField] Button m_ServerButton;
-        [SerializeField] Button m_HostButton;
-        [SerializeField] Button m_ClientButton;
-        [SerializeField] Button m_DisconnectButton;
-        [SerializeField] Button m_QuitSceneButton;
-        [SerializeField] Color32 m_HighlightedButtonColor;
-        [SerializeField] Color32 m_DisabledButtonColor;
-        [SerializeField] Color32 m_ButtonOutlineColor;
-        [SerializeField] Color32 m_ButtonOutlineHighlightedColor;
-        [SerializeField] Color32 m_ButtonOutlineDisabledColor;
+        const string k_DefaultIP = "127.0.0.1";
+        const string k_DefaultServerListenAddress = "0.0.0.0"; //note: this is not safe for real world usage and would limit you to IPv4-only addresses, but this goes out the scope of this sample.
+        const ushort k_DefaultPort = 7979;
+
+        VisualElement m_Root;
+        TextField m_AddressInputField;
+        TextField m_PortInputField;
+        Button m_ServerButton;
+        Button m_HostButton;
+        Button m_ClientButton;
+        Button m_DisconnectButton;
+        Button m_QuitSceneButton;
+
         [SerializeField] string m_SelectionScreenSceneName;
         [SerializeField] GameObject m_ServerOnlyOverlay;
-        Button[] m_Buttons;
 
         void Awake()
         {
-            m_ServerButton.onClick.AddListener(StartServer);
-            m_HostButton.onClick.AddListener(StartHost);
-            m_ClientButton.onClick.AddListener(StartClient);
-            m_DisconnectButton.onClick.AddListener(Disconnect);
-            m_QuitSceneButton.onClick.AddListener(QuitScene);
-            m_Buttons = new Button[] { m_ServerButton, m_HostButton, m_ClientButton };
+            var uiDocument = GetComponent<UIDocument>();
+            m_Root = uiDocument.rootVisualElement;
+
+            m_AddressInputField = m_Root.Q<TextField>("IPAddressField");
+            m_AddressInputField.SetValueWithoutNotify(k_DefaultIP);
+
+            m_PortInputField = m_Root.Q<TextField>("PortField");
+            m_PortInputField.SetValueWithoutNotify(k_DefaultPort.ToString());
```

Comment:
A callback needs to be reigstered to those as well, in order to validate the input to be sure people don't write stuff like "pizza" in the IP address field.

1. IP should follow the IPV4 structure (check [this comment](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/pull/116#discussion_r1149579461) to see how to do it)
2. Port should be an integer number between 0 and 65535